Compilation fails with error: no matching function for call to 'std::function<const StringsBundle&()>::function(bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>)'

**Describe the issue**
I tried to compile Organic Maps with `cmake .. -GNinja -DWITH_SYSTEM_PROVIDED_3PARTY=yes` and all of the patches of #6801.
It works for all targets (including the `OMaps` binary and the program runs well) but two targets fail:
- `map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o`
- `map_tests` (?) this step target depends on the other one and is never actually build

The error message of the compiler begins like that (the error repeats a few times):
```
[1/2] /usr/bin/ccache /usr/bin/ccache /usr/bin/c++ -DOMIM_UNIT_TEST_WITH_QT_EVENT_LOOP -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_POSITIONING_LIB -DQT_WIDGETS_LIB -DRELEASE -DUSE_FILE32API -I/home/gerion/src/organicmaps -I/home/gerion/src/organicmaps/build/3party/gflags/include -I/home/gerion/src/organicmaps/3party/liboauthcpp/include -I/home/gerion/src/organicmaps/3party/protobuf/protobuf/src -I/home/gerion/src/organicmaps/3party/opening_hours/. -I/home/gerion/src/organicmaps/3party/glm -I/home/gerion/src/organicmaps/3party/vulkan_wrapper/. -I/home/gerion/src/organicmaps/3party/harfbuzz/harfbuzz/src -isystem /home/gerion/src/organicmaps/3party/boost -isystem /home/gerion/src/organicmaps/3party/utf8cpp/include -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtPositioning -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtGui -isystem /usr/include/freetype2 -O3 -DNDEBUG -std=c++17 -fPIE -fdiagnostics-color=always -ffast-math -Ofast -Wall -Wextra -Wpedantic -Wno-unused-parameter -fPIC -MD -MT map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o -MF map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o.d -o map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o -c /home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp
FAILED: map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o 
/usr/bin/ccache /usr/bin/ccache /usr/bin/c++ -DOMIM_UNIT_TEST_WITH_QT_EVENT_LOOP -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_POSITIONING_LIB -DQT_WIDGETS_LIB -DRELEASE -DUSE_FILE32API -I/home/gerion/src/organicmaps -I/home/gerion/src/organicmaps/build/3party/gflags/include -I/home/gerion/src/organicmaps/3party/liboauthcpp/include -I/home/gerion/src/organicmaps/3party/protobuf/protobuf/src -I/home/gerion/src/organicmaps/3party/opening_hours/. -I/home/gerion/src/organicmaps/3party/glm -I/home/gerion/src/organicmaps/3party/vulkan_wrapper/. -I/home/gerion/src/organicmaps/3party/harfbuzz/harfbuzz/src -isystem /home/gerion/src/organicmaps/3party/boost -isystem /home/gerion/src/organicmaps/3party/utf8cpp/include -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtPositioning -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtGui -isystem /usr/include/freetype2 -O3 -DNDEBUG -std=c++17 -fPIE -fdiagnostics-color=always -ffast-math -Ofast -Wall -Wextra -Wpedantic -Wno-unused-parameter -fPIC -MD -MT map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o -MF map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o.d -o map/map_tests/CMakeFiles/map_tests.dir/bookmarks_test.cpp.o -c /home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp
In file included from /home/gerion/src/organicmaps/map/framework.hpp:5,
                 from /home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp:9:
/home/gerion/src/organicmaps/map/bookmark_manager.hpp: In instantiation of 'BookmarkManager::Callbacks::Callbacks(StringsBundleProvider&&, SearchAPIProvider&&, CreateListener&&, UpdateListener&&, DeleteListener&&, AttachListener&&, DetachListener&&) [with StringsBundleProvider = bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>; SearchAPIProvider = std::function<SearchAPI&()>; CreateListener = std::function<void(const std::vector<BookmarkInfo>&)>; UpdateListener = std::function<void(const std::vector<BookmarkInfo>&)>; DeleteListener = std::function<void(const std::vector<long unsigned int, std::allocator<long unsigned int> >&)>; AttachListener = std::function<void(const std::vector<BookmarkGroupInfo>&)>; DetachListener = std::function<void(const std::vector<BookmarkGroupInfo>&)>]':
/home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp:195:41:   required from here
/home/gerion/src/organicmaps/map/bookmark_manager.hpp:90:9: error: no matching function for call to 'std::function<const StringsBundle&()>::function(bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>)'
   90 |       : m_getStringsBundle(std::forward<StringsBundleProvider>(stringsBundleProvider))
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/functional:59,
                 from /home/gerion/src/organicmaps/testing/testregister.hpp:3,
                 from /home/gerion/src/organicmaps/testing/testing.hpp:3,
                 from /home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp:1:
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:435:9: note: candidate: 'template<class _Functor, class _Constraints> std::function<_Res(_ArgTypes ...)>::function(_Functor&&) [with _Constraints = _Functor; _Res = const StringsBundle&; _ArgTypes = {}]'
  435 |         function(_Functor&& __f)
      |         ^~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:435:9: note:   template argument deduction/substitution failed:
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/move.h:37,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/stl_function.h:60,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/functional:49:
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/type_traits: In substitution of 'template<bool _Cond, class _Tp> using std::__enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = void]':
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:353:8:   required by substitution of 'template<class _Res, class ... _ArgTypes> template<class _Cond, class _Tp> using std::function<_Res(_ArgTypes ...)>::_Requires = std::__enable_if_t<_Cond::value, _Tp> [with _Cond = std::function<const StringsBundle&()>::_Callable<bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>, bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>, std::__invoke_result<bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>&> >; _Tp = void; _Res = const StringsBundle&; _ArgTypes = {}]'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:434:9:   required from 'BookmarkManager::Callbacks::Callbacks(StringsBundleProvider&&, SearchAPIProvider&&, CreateListener&&, UpdateListener&&, DeleteListener&&, AttachListener&&, DetachListener&&) [with StringsBundleProvider = bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>; SearchAPIProvider = std::function<SearchAPI&()>; CreateListener = std::function<void(const std::vector<BookmarkInfo>&)>; UpdateListener = std::function<void(const std::vector<BookmarkInfo>&)>; DeleteListener = std::function<void(const std::vector<long unsigned int, std::allocator<long unsigned int> >&)>; AttachListener = std::function<void(const std::vector<BookmarkGroupInfo>&)>; DetachListener = std::function<void(const std::vector<BookmarkGroupInfo>&)>]'
/home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp:195:41:   required from here
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/type_traits:116:11: error: no type named 'type' in 'struct std::enable_if<false, void>'
  116 |     using __enable_if_t = typename enable_if<_Cond, _Tp>::type;
      |           ^~~~~~~~~~~~~
/home/gerion/src/organicmaps/map/bookmark_manager.hpp: In instantiation of 'BookmarkManager::Callbacks::Callbacks(StringsBundleProvider&&, SearchAPIProvider&&, CreateListener&&, UpdateListener&&, DeleteListener&&, AttachListener&&, DetachListener&&) [with StringsBundleProvider = bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>; SearchAPIProvider = std::function<SearchAPI&()>; CreateListener = std::function<void(const std::vector<BookmarkInfo>&)>; UpdateListener = std::function<void(const std::vector<BookmarkInfo>&)>; DeleteListener = std::function<void(const std::vector<long unsigned int, std::allocator<long unsigned int> >&)>; AttachListener = std::function<void(const std::vector<BookmarkGroupInfo>&)>; DetachListener = std::function<void(const std::vector<BookmarkGroupInfo>&)>]':
/home/gerion/src/organicmaps/map/map_tests/bookmarks_test.cpp:195:41:   required from here
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:404:7: note: candidate: 'std::function<_Res(_ArgTypes ...)>::function(std::function<_Res(_ArgTypes ...)>&&) [with _Res = const StringsBundle&; _ArgTypes = {}]'
  404 |       function(function&& __x) noexcept
      |       ^~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:404:27: note:   no known conversion for argument 1 from 'bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>' to 'std::function<const StringsBundle&()>&&'
  404 |       function(function&& __x) noexcept
      |                ~~~~~~~~~~~^~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:386:7: note: candidate: 'std::function<_Res(_ArgTypes ...)>::function(const std::function<_Res(_ArgTypes ...)>&) [with _Res = const StringsBundle&; _ArgTypes = {}]'
  386 |       function(const function& __x)
      |       ^~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:386:32: note:   no known conversion for argument 1 from 'bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>' to 'const std::function<const StringsBundle&()>&'
  386 |       function(const function& __x)
      |                ~~~~~~~~~~~~~~~~^~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:375:7: note: candidate: 'std::function<_Res(_ArgTypes ...)>::function(std::nullptr_t) [with _Res = const StringsBundle&; _ArgTypes = {}; std::nullptr_t = std::nullptr_t]'
  375 |       function(nullptr_t) noexcept
      |       ^~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:375:16: note:   no known conversion for argument 1 from 'bookmarks_test::UnitClass_Runner_Bookmarks_ImportKML::Bookmarks_ImportKML()::<lambda()>' to 'std::nullptr_t'
  375 |       function(nullptr_t) noexcept
      |                ^~~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:368:7: note: candidate: 'std::function<_Res(_ArgTypes ...)>::function() [with _Res = const StringsBundle&; _ArgTypes = {}]'
  368 |       function() noexcept
      |       ^~~~~~~~
/usr/lib/gcc/x86_64-pc-linux-gnu/13/include/g++-v13/bits/std_function.h:368:7: note:   candidate expects 0 arguments, 1 provided
```

**Steps to reproduce**
1. Apply PR #6801
2. `mkdir biuld; cd build; cmake .. -GNinja -DWITH_SYSTEM_PROVIDED_3PARTY=yes; ninja`


**Expected behaviour**
All targets should compile just fine.


**System information:**
 - Operating system and its version: Gentoo Linux (kernel 6.6.2, GCC 13.2.1)
 - Organic Maps version: Git HEAD
 - Device Model: should not be relevant here

--- a/map/map_tests/bookmarks_test.cpp
+++ b/map/map_tests/bookmarks_test.cpp
@@ -134,9 +134,9 @@ char const * kmlString =
     "</kml>";
 
 #define BM_CALLBACKS {                                                            \
-    []()                                                                         \
+    []() -> StringsBundle const &                                                \
     {                                                                            \
-      static StringsBundle dummyBundle;                                          \
+      static StringsBundle const dummyBundle;                                    \
       return dummyBundle;                                                        \
     },                                                                           \
     static_cast<BookmarkManager::Callbacks::GetSeacrhAPIFn>(nullptr),            \
@@ -1216,9 +1216,9 @@ UNIT_CLASS_TEST(Runner, Bookmarks_Listeners)
   };
 
   BookmarkManager::Callbacks callbacks(
-    []()
+    []() -> StringsBundle const &
     {
-      static StringsBundle dummyBundle;
+      static StringsBundle const dummyBundle;
       return dummyBundle;
     },
     static_cast<BookmarkManager::Callbacks::GetSeacrhAPIFn>(nullptr),

